src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,40 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todos: Todo[];
+  selectTodo: (value: number) => number | void;
+  selectedTodoId: number;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todos,
+  selectTodo,
+  selectedTodoId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const foundTodo = todos.find(todo => todo.id === selectedTodoId);
+
+  useEffect(() => {
+    if (foundTodo) {
+      getUser(foundTodo.userId)
+        .then(response => setUser(response));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code